source/change-streams/change-streams.rst

```diff
@@ -43,24 +43,56 @@ Resumable Error
 
 An error is considered resumable if it meets any of the following criteria:
 
-- any error encountered which is not a server error (e.g. a timeout error or
+- Any error encountered which is not a server error (e.g. a timeout error or
   network error)
 
-- *any* server error response from a getMore command excluding those
-  containing the error label `NonResumableChangeStreamError` and those
-  containing the following error codes
+- For servers with wire version 9 or higher (server version 4.4 or higher), any
+  server error with the `ResumableChangeStreamError` error label.
+
+- For servers with wire version less than 9, a server error with one of the
+  following codes:
 
   .. list-table::
     :header-rows: 1
 
     * - Error Name
       - Error Code
-    * - Interrupted
-      - 11601
-    * - CappedPositionLost
-      - 136
-    * - CursorKilled
-      - 237
+    * - HostUnreachable
```

Comment:
This list was generated based on the server changes made in https://github.com/mongodb/mongo/commit/3d821c25e2944668b7359a0bf6e586cc8625b9a2#diff-cf7105bb60d0308203e7ef6ec12e07c3R74 (SERVER-45505).

Reply:
Nice. Verified this list is correct by going through error_codes.yml:
https://gist.github.com/kevinAlbs/c5eeead90dca71097ee111530f5b1704

Reply:
Also, questions for other reviewers. As @divjotarora pointed out in the code review for SERVER-45505, the server adds ResumableChangeStreamError to replies of aggregate as well. Copying his comment here:

> The current drivers specification
> (https://github.com/mongodb/specifications/blob/master/source/change-streams/c...)
> explicitly states that all errors on aggregate are not resumeable errors. The
> operations layer of the driver will retry the aggregate once for some subset of
> errors (e.g. network failures, NotMaster errors) per the retryable reads
> specification, but if both the original and retry error, the error will be
> reported up to the change stream code, which won't look for this label to figure
> out if another aggregate should be sent. I believe this was done because it does
> not make sense to "resume" a change stream that hasn't been established in the
> first place.

Does it seem reasonable to resume on aggregate if it contains that label? If so, I can create a separate spec ticket. I don't think we should do it as part of this spec ticket, since it would be a non-trivial additional change, and outside of the scope of the problem this spec ticket solves.

Reply:
It's going to require significant refactoring in some drivers to be able to resume aggregate.  I would prefer that the server, instead of labelling an aggregate response with a ResumableChangeStreamError, instead just returned a response with no documents in firstBatch, and then failed the first getMore.

But if that turns out to not be feasible, we'll have to take this on as new work.